main.go

```diff
@@ -421,124 +400,139 @@ func mainCore() error {
+		// StakeDatabase
+		wg.Add(2)
+		go sdbChainMonitor.BlockConnectedHandler()
+		go sdbChainMonitor.ReorgHandler()
 
-	if usePG {
 		// dcrsqlite does not handle new blocks except during reorg
 		wg.Add(2)
-		go auxDBChainMonitor.BlockConnectedHandler()
-		go auxDBChainMonitor.ReorgHandler()
-	}
-
-	if cfg.MonitorMempool {
-		mpoolCollector := mempool.NewMempoolDataCollector(dcrdClient, activeChain)
-		if mpoolCollector == nil {
-			return fmt.Errorf("Failed to create mempool data collector")
+		go wiredDBChainMonitor.BlockConnectedHandler()
+		go wiredDBChainMonitor.ReorgHandler()
+
+		if usePG {
+			// dcrsqlite does not handle new blocks except during reorg
+			wg.Add(2)
+			go auxDBChainMonitor.BlockConnectedHandler()
+			go auxDBChainMonitor.ReorgHandler()
 		}
 
-		mpData, err := mpoolCollector.Collect()
-		if err != nil {
-			return fmt.Errorf("Mempool info collection failed while gathering"+
-				" initial data: %v", err.Error())
-		}
+		if cfg.MonitorMempool {
+			mpoolCollector := mempool.NewMempoolDataCollector(dcrdClient, activeChain)
+			if mpoolCollector == nil {
+				return fmt.Errorf("Failed to create mempool data collector")
+			}
 
-		// Store initial MP data
-		if err = baseDB.MPC.StoreMPData(mpData, time.Now()); err != nil {
-			return fmt.Errorf("Failed to store initial mempool data (wiredDB): %v",
-				err.Error())
-		}
+			mpData, err := mpoolCollector.Collect()
+			if err != nil {
+				return fmt.Errorf("Mempool info collection failed while gathering"+
+					" initial data: %v", err.Error())
+			}
+
+			// Store initial MP data
+			if err = baseDB.MPC.StoreMPData(mpData, time.Now()); err != nil {
+				return fmt.Errorf("Failed to store initial mempool data (wiredDB): %v",
+					err.Error())
+			}
+
+			// Setup monitor
+			mpi := &mempool.MempoolInfo{
+				CurrentHeight:               mpData.GetHeight(),
+				NumTicketPurchasesInMempool: mpData.GetNumTickets(),
+				NumTicketsSinceStatsReport:  0,
+				LastCollectTime:             time.Now(),
+			}
 
-		// Setup monitor
-		mpi := &mempool.MempoolInfo{
-			CurrentHeight:               mpData.GetHeight(),
-			NumTicketPurchasesInMempool: mpData.GetNumTickets(),
-			NumTicketsSinceStatsReport:  0,
-			LastCollectTime:             time.Now(),
+			newTicketLimit := int32(cfg.MPTriggerTickets)
+			mini := time.Duration(cfg.MempoolMinInterval) * time.Second
+			maxi := time.Duration(cfg.MempoolMaxInterval) * time.Second
+
+			mpm := mempool.NewMempoolMonitor(mpoolCollector, mempoolSavers,
+				notify.NtfnChans.NewTxChan, quit, &wg, newTicketLimit, mini, maxi, mpi)
+			wg.Add(1)
+			go mpm.TxHandler(dcrdClient)
 		}
+		return nil
+	}
 
-		newTicketLimit := int32(cfg.MPTriggerTickets)
-		mini := time.Duration(cfg.MempoolMinInterval) * time.Second
-		maxi := time.Duration(cfg.MempoolMaxInterval) * time.Second
+	// do not collect the data summary if the sync status page has been activated.
+	if !explore.SyncStatus {
+		if err := collectBlocksAfterSyncing(); err != nil {
```

Comment:
How can calling this function work before `getSyncd` is run?

Reply:
First, if all the event handlers are started, and then sync is run, there's trouble when an event comes and is processed for a block beyond the DB best block.  That's why we don't start handlers until later.
Second, the Collect()/Store() parts of `collectBlocksAfterSyncing` that get data for the explorer are for an old block.  How is it going to get the fresh data when sync is done (or as it goes on)?

Reply:
I expect it to work especially in the case scenarios when its called before `getSyncd`, I mean the times when `explore.SyncStatus` is false and sync status page is not running as the only accessible page on the web server. The number of blocks behind the best block then should be quite few (can only be set to a max of 5000) which means that by the time the system expects to handle this unsynced load, much of the data may have already been loaded.

"quite few" in the above statement is not always correct because some machines will take longer to process a given load. This brings us to the option of configuring the `sync-status-limit` value that should help us control what load a given machine can process simultaneously while serving the web server an still be effective.

Reply:
> the times when explore.SyncStatus is false and sync status page is not running as the only accessible page on the web server

I get that in the above case the idea is to make then entire explorer available.  Correct?

The problems I mentioned are still a concern.  The new block and reorg handlers are up and running because of `collectBlocksAfterSyncing`, which means there will be new block notifications from dcrd for new blocks that we cannot possibly connect in dcrdata's DBs.  The handlers should not be running for this reason.

Also, when `collectBlocksAfterSyncing` runs Collect and Store, what block is it collecting and sending to the explorer UI? It will not be in the DB. Also, when sync does finish, it won't run Collect and Store after sync because it's moved up here now...

Reply:
>I get that in the above case the idea is to make then entire explorer available. Correct?

Yes

> The new block and reorg handlers are up and running because of collectBlocksAfterSyncing, which means there will be new block notifications from dcrd for new blocks that we cannot possibly connect in dcrdata's DBs. 

I get your concern and you correct on this `The handlers should not be running for this reason`. I will fix that so that the handlers will be started still in the `collectBlocksAfterSyncing` function but will only trigger any further processing if the data that is needed already exist in all dbs.

The reason why i think the handlers should remain in the same position as they are, is that  some machines will be fast enough to process the "quite few" blocks that are pending.

I also need to think through this section or the again as it too error prone.

Reply:
Sure, some machine may get done processing the pending blocks before a new one is mined, but we can't rely on that.

Reply:
Generally for all machines whether fast or slow, the handlers will have to confirm if the needed information already exists. This means that monitors will not trigger further processing if the need block is not yet processed.

Reply:
Just to be clear, we're not talking about when dcrd is still syncing, this is regarding when dcrdata is syncing to dcrd.  Take `BlockConnectedHandler` in the `blockdata` package.  Say sync is at block 1000, and dcrd sends a notification that block 1200 was just mined, then `blockdata`'s `BlockConnectedHandler` gets the block hash in a message, collects all the data that dcrd has about the block without problem since dcrd has it, but the problems come with when `BlockConnectedHandler` calls `CollectHash`, which calls `blockdata/(*Collector).CollectBlockInfo`, for because stakedb is behind and there's no ticket pool info for that new block from dcrd.  If there were a workaround to that there would still be a problem when `BlockConnectedHandler` calls `Store` for all the `dataSaver`s.

Reply:
This is another valid concern and I think any data update triggered by dcrd should be blocked by all means while the sync is in progress. What we could allow is that data that has been pushed to the dbs during sync could be processed for display as the latest update fetched. This is data processing that I was taking about with this.
>the handlers will have to confirm if the needed information already exists.

Also the sync code is designed to stop after initial startup  if the best block from dcrd is in sync with the best block from dcrdata. So the blocked dcrd updates will be handled eventually.

BTW, this PR only affects how dcrdata initial startup will work after that the current norm takes precedence.

Reply:
> What we could allow is that data that has been pushed to the dbs during sync could be processed for display as the latest update fetched.

I referred to this earlier, and I would suggest updating the explorer UI every ~20 blocks while syncing to dcrd.

>  So the blocked dcrd updates will be handled eventually.

While dcrd is syncing to the network, each block processed by dcrd will be notified to dcrdata.  They will stack up, yes, but the regular sync loop would have handled them.  They'll be out of order w.r.t. the DB height and/or already processed.

Reply:
As long as the first sync to all tables except to the addresses table isn't done yet this 

>While dcrd is syncing to the network, each block processed by dcrd will be notified to dcrdata. They will stack up, yes, but the regular sync loop would have handled them. They'll be out of order w.r.t. the DB height and/or already processed.

will be handled here.

```go
// Update height, the end condition for the loop
if nodeHeight, err = client.NodeHeight(); err != nil {
	return ib, fmt.Errorf("GetBestBlock failed: %v", err)
}
```
if its done it will definately be picked up this immediately after the addresses table sync.

```go
for baseDBHeight < height {
	fetchToHeight = auxDBHeight + 1
	baseDBHeight, _, err = getSyncd(updateAllAddresses, updateAllVotes,
			newPGIndexes, fetchToHeight)
	if err != nil {
		return err
	}
	_, height, err = dcrdClient.GetBestBlock()
	if err != nil {
		return fmt.Errorf("unable to get block from node: %v", err)
	}
}
```

Reply:
>How can calling this function work before getSyncd is run?

It now makes lots of sense why initiating the monitors before sync is complete isn't such a good idea. If I went ahead with my implementation of initiating the monitors & handlers before, I would be forced to block them all to avoid db corruption thereby piling unnecessarily pressure on the underlying hardware's processing power.